From be8735d31f0d9cc42f5a409c09ccdf08354e3e65 Mon Sep 17 00:00:00 2001
From: Mariam John <mariamj@us.ibm.com>
Date: Mon, 22 Nov 2021 13:15:23 -0600
Subject: [release-1.12] revert /host prefix from CNI log volume mount path.
 (#36038) (#36098)

* Add securityContext to CNI DaemonSet to run as root. (#35986)

* Add securityContext to CNI DaemonSet to run as root.

* clean.

* update.

(cherry picked from commit 2ea98aae52a237229637660485c12affb1eff14d)

* revert /host prefix from CNI log volume mount path. (#36038)

(cherry picked from commit 8dcf53141d6d50efef56da1506b3d1076a93450a)

Co-authored-by: Pengyuan Bian <bianpengyuan@google.com>
---
 manifests/charts/istio-cni/templates/daemonset.yaml | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/manifests/charts/istio-cni/templates/daemonset.yaml b/manifests/charts/istio-cni/templates/daemonset.yaml
index f24782f30c..fb83dc76be 100644
--- a/manifests/charts/istio-cni/templates/daemonset.yaml
+++ b/manifests/charts/istio-cni/templates/daemonset.yaml
@@ -78,6 +78,10 @@ spec:
             httpGet:
               path: /readyz
               port: 8000
+          securityContext:
+            runAsGroup: 0
+            runAsUser: 0
+            runAsNonRoot: false
           command: ["install-cni"]
           env:
 {{- if .Values.cni.cniConfFileName }}
@@ -141,6 +145,10 @@ spec:
           imagePullPolicy: {{ .Values.cni.pullPolicy | default .Values.global.imagePullPolicy }}
 {{- end }}
           command: ["/opt/local/bin/istio-cni-taint"]
+          securityContext:
+            runAsUser: 1337
+            runAsGroup: 1337
+            runAsNonRoot: true
           env:
           - name: "TAINT_RUN-AS-DAEMON"
             value: "true"
-- 
2.35.3

